main.cpp

```diff
@@ -182,9 +227,9 @@ void *memalign(size_t a, size_t n) {
     i = i + sizeof(size_t);
     i = ROUNDUP(i, a);
     j = ROUNDUP(i + m, MAGIC_GRAN);
-    if (j > mag->capacity) {
+    //if (j > mag->capacity) {
```

Comment:
This line surprises me and I suppose it relates to why this is still a draft. Since wouldn't this cause us to blow away and recreate mappings?

Reply:
Oh, most likely it is the draft code. I just was trying out different things depending on my intuition of what's happening, not sure how it got into PR